### DISCUSSION TITLE
Legacy Silverlight on today's Azure

### DISCUSSION THREAD

**Discussion by kellyethridge** [MarimerLLC/csla#4379] [Questions] [answered]
Hello,

We have a legacy Silverlight application using CSLA 4.5.701 that still needs to function. It's been working fine for 10 years on a Windows 2012 R2 server. We've been migrating our products to a Windows Server 2022 in Azure. I have been unsuccessful at getting the Silverlight client to reach the WCF endpoint it seems. I'm getting the old "The remote server returned an error: NotFound" message. I'm wondering if anyone has had any guidance getting Silverlight (more like the WCF protocol) to work in the Azure world?

Thank you

**Comment by rockfordlhotka** [comment]
It is _maybe_ that TLS 1.0 and 1.1 are now obsolete and may no longer be allowed on modern operating systems, but Silverlight might still use it?

You might consider using [OpenSilver](https://opensilver.net/) to modernize your app so it is still "Silverlight", but on modern software.

**Reply by kellyethridge** [reply]
Thanks for the suggestion. I enabled both 1.0 and 1.1 to no avail. Is one of those necessary for the HTTP WCF channel to work? From what I can tell it works through port 443 on the server. I didn't keep my WCF skills up, so I'm kind of just looking at stuff.

Thanks for the OpenSilver suggestion, however, we've been moving away from Silverlight. We're at the end of migration except for a few pieces. Unfortunately it didn't align with the server migration, so now here we are. I feel like it's 2010 all over :)

**Comment by hurcane** [comment, marked as answer]
I'm not a WCF expert, but you might check out the steps notated in this Microsoft forum. It includes Powerahell commands, which are often preferred when dealing with Azure resources.

https://answers.microsoft.com/en-us/windowserver/forum/all/we-have-a-net-48-wcf-application-on-windows-server/575248ab-c8c2-4afd-92cc-216c883a7dda

**Reply by kellyethridge** [reply]
Yes, that fixed the issue. Thank you for your help!
